backend/src/test/java/de/silvia/backend/security/services/RegisterServiceTest.java

```diff
@@ -0,0 +1,40 @@
+package de.silvia.backend.security.services;
+
+import de.silvia.backend.api.UserDto;
+import de.silvia.backend.security.models.User;
+import de.silvia.backend.security.repositories.IUserRepo;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+class RegisterServiceTest {
+
+    @Mock
+    IUserRepo userRepo;
+
+    @Mock
+    PasswordEncoder encoder;
+
+    @Test
+    public void registerUser(){
+        assertNotNull(userRepo);
+        UserDto mockedUserDto = new UserDto("Julius", "Schmidt", "schmitti",
+                "123456", "schmitt@uli.de");
+        String encodedPw = encoder.encode(mockedUserDto.getPassword());
+        User mockedUser = User.newUser(mockedUserDto.getFirstName(), mockedUserDto.getEmail(),
+                mockedUserDto.getLastName(), mockedUserDto.getUserName(),
+                encodedPw, List.of(
+                        new SimpleGrantedAuthority("API_READWRITE")));
```

Comment:
I made a DTO because the original function takes that as input. As I have understood the tests, I should simulate the original functions. The DTO comes from the frontend, which is only simulated here.